packages/fxa-settings/src/lib/model-data/model-data-provider.ts

```diff
@@ -158,10 +171,55 @@ export class ModelDataProvider {
 }
 
 export class ModelValidationErrors extends Error {
-  constructor(public readonly errors: ValidationError[]) {
+  public condition: '' | 'QueryParameterValidation';
+
+  constructor(
+    public readonly errors: ValidationError[],
+    public readonly keyLookup: Record<string, string> = {}
+  ) {
     super(
       'Model Validation Errors Encountered! Fields:' +
         errors.map((x) => x.toString()).join(',')
     );
+    this.condition = '';
+  }
+
+  /**
+   * Converts the error to a user friendly list of issues that can
```

Comment:
I'm not sure these should be shown to the user if there's nothing they can do about the error?